src/models/Project/addLocaleToProject/index.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+import Project from '@models/Project'
+import Locale from '@models/Locale'
+import type { ValidationError } from '@types/ValidationError'
+
+export type AddLocaleToProjectArgs = {
+  projectId: string,
+  localeId: string,
+  userId: string
+}
+
+type AddProjectToLocaleResponse = {
+  project?: Project,
+  errors: Array<ValidationError>
+}
+
+const addLocaleToProject = async ({
+  projectId,
+  localeId,
+  userId
+}: AddLocaleToProjectArgs): Promise<AddProjectToLocaleResponse> => {
+  const errors: Array<ValidationError> = []
+
+  const project = await Project.query()
+    .eager('locales')
+    .findOne({ id: projectId, ownerId: userId })
+  const locale = await Locale.query().findById(localeId)
+
+  if (!project || !locale) {
+    const notFound = !project ? 'project' : 'locale'
+    errors.push({
+      field: notFound,
+      message: `This ${notFound} could not be found.`
+    })
+
+    return { errors }
+  }
+
+  project.locales.map(({ id }) => {
```

Comment:
I think we should do a count using the database instead of this map. Also, prefer `forEach` when you’re not using the return value of `.map(…)`.
Or maybe there is a `insertIfNotExists`?